src/CMakeLists.txt

```diff
@@ -204,11 +208,12 @@ if(MSVC)
   target_compile_options(spatialindex PRIVATE /wd4068)
   target_compile_options(spatialindex_c PRIVATE /wd4068)
 
-  target_compile_definitions(spatialindex_c PRIVATE -DSIDX_DLL_EXPORT=1)
-  target_compile_definitions(spatialindex PRIVATE -DSIDX_DLL_EXPORT=1)
-  if(NOT WITH_STATIC_SIDX)
-    target_compile_definitions(spatialindex PRIVATE -DSIDX_DLL_IMPORT=1)
-    target_compile_definitions(spatialindex_c PRIVATE -DSIDX_DLL_IMPORT=1)
+ if (BUILD_SHARED_LIBS)
+   target_compile_definitions(spatialindex PRIVATE -DSIDX_DLL_EXPORT=1)
+   target_compile_definitions(spatialindex_c PRIVATE -DSIDX_DLL_EXPORT=1)
+ else()
+  target_compile_definitions(spatialindex PRIVATE -DSIDX_DLL_IMPORT=1)
+  target_compile_definitions(spatialindex_c PRIVATE -DSIDX_DLL_IMPORT=1)
```

Comment:
Awesome to remove custom `WITH_STATIC_SIDX`, as a global `BUILD_SHARED_LIBS` is preferred.

Two minor nits about style: no space after the `if` command, and use a multiple of two space indents (not one).